screens/recipes/SingleRecipeScreen.js

```diff
@@ -26,6 +30,27 @@ function Recipe(props) {
     });
   }, [navigation, showMenu, recipe.attributes.name]);
 
+  function ingredientInfo(ingredient) {
+    return {
+      id: ingredient.id,
+      name: ingredient.attributes.ingredient.name,
+      price: ingredient.attributes.ingredient.price,
+      currentPrice: ingredient.attributes.ingredient.price * ingredient.attributes.ingredient_quantity /
+        ingredient.attributes.ingredient.quantity,
+      unitQuantity: ingredient.attributes.ingredient.quantity,
+      recipeQuantity: ingredient.attributes.ingredient_quantity,
+      measure: ingredient.attributes.ingredient.measure,
+    };
+  }
+
+  useEffect(() => {
+    const currentIngredients = [];
+    recipe.attributes.recipe_ingredients.data.forEach((ingredient) => {
+      currentIngredients.push(ingredientInfo(ingredient));
+    });
+    setIngredients(currentIngredients);
```

Comment:
Pueden usar `map` para esto! Así:

```suggestion
    setIngredients(
      recipe.attributes.recipe_ingredients.data.map(
        (ingredient) => ingredientInfo(ingredient),
      ),
    );
```

Creo que queda más limpio! Avísenme qué opinan

Reply:
oka!

Reply:
Gracias por la sugerencia! Lo seguiré haciendo así